app/services/NotificationService.scala

```diff
@@ -116,19 +118,24 @@ class NotificationService @Inject()(val amlsNotificationConnector: AmlsNotificat
       .getDeclaredConstructor().newInstance().asInstanceOf[{ def endDate(contactType: ContactType, endDate: String, url: String, referenceNumber: String): String }]
 
     amlsNotificationConnector.getMessageDetailsByAmlsRegNo(amlsRegNo, id, accountTypeId) map {
-      case Some(notificationDetails) => {
+      case Some(notificationDetails) =>
         for {
           message <- notificationDetails.messageText
           details <- NotificationDetails.convertEndDateMessageText(message)
         } yield {
+
+          val endDate: String =
+          if(templateVersion == "v5m0" && (contactType == ContactType.RenewalReminder || contactType == ContactType.NewRenewalReminder))
```

Comment:
Why does this have to be done especially for v5m0, but not for others, please?  Perhaps because content change has mandated it.  I would really prefer that it is dealt somewhere when the content is different for each version case.  

In any case, as above, it is not safe: a future v6 will revert to pre-v4 behaviour

Reply:
This has too be done only for v5 because the date format in the inbox message has changed from YYYY-MM-DD to DD-MM-YYYY for renewal reminder and it shouldn't change for the old templates. This is  done here as this is the place we are passing end date in the template

Reply:
Let me have a look if that can be done in MessageDetails.scala 👍

Reply:
date change is now done at MessageDetails.scala for the two contactTYpe
https://github.com/hmrc/amls-frontend/commit/dbc622a525963869c2a520ceceab6155b8a8c2dc